src/ActionList/Item.tsx

```diff
@@ -393,35 +394,31 @@ export function Item(itemProps: Partial<ItemProps> & {item?: ItemInput}): JSX.El
           <LeadingVisual />
         </LeadingVisualContainer>
       )}
-      <StyledItemContent>
+      <StyledItemContent descriptionVariant={descriptionVariant}>
         {children}
-        {(text || description) && (
-          <TextContainer descriptionVariant={descriptionVariant}>
-            {text && <div>{text}</div>}
-            {description && (
-              <DescriptionContainer descriptionVariant={descriptionVariant}>
-                {descriptionVariant === 'block' ? (
-                  description
-                ) : (
-                  <Truncate title={description} inline={true} maxWidth="100%">
-                    {description}
-                  </Truncate>
-                )}
-              </DescriptionContainer>
-            )}
-          </TextContainer>
-        )}
-        {(TrailingIcon || trailingText) && (
-          <TrailingVisualContainer variant={variant} disabled={disabled}>
-            {trailingText && <div>{trailingText}</div>}
-            {TrailingIcon && (
-              <div>
-                <TrailingIcon />
-              </div>
+        {text ? <TextContainer id={`${id}-label`}>{text}</TextContainer> : null}
+        {description ? (
+          <DescriptionContainer id={`${id}-description`} descriptionVariant={descriptionVariant}>
+            {descriptionVariant === 'block' ? (
+              description
+            ) : (
+              <Truncate title={description} inline={true} maxWidth="100%">
+                {description}
+              </Truncate>
             )}
-          </TrailingVisualContainer>
-        )}
+          </DescriptionContainer>
+        ) : null}
       </StyledItemContent>
+      {(TrailingIcon || trailingText) && (
+        <TrailingVisualContainer variant={variant} disabled={disabled}>
+          {trailingText && <div>{trailingText}</div>}
+          {TrailingIcon && (
+            <div>
+              <TrailingIcon />
+            </div>
+          )}
+        </TrailingVisualContainer>
+      )}
```

Comment:
I think moving this outside of `StyledItemContent` will cause the item divider lines to end after the text/description.  Is that desired?

Reply:
Good catch! That is not desired. 👍

Reply:
Fixed in https://github.com/primer/components/pull/1320/commits/408ff2473c40a867b41c8bbedc1cf9499fc677ae, but that commit may have introduced other issues. I haven’t found any regressions, but I’d appreciate more eyes on it.